src/transaction/input/scripthash.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+var _ = require('lodash');
+var inherits = require('inherits');
+var Input = require('./input');
+var Output = require('../output');
+var $ = require('../../util/preconditions');
+
+var Script = require('../../script');
+var Signature = require('../../crypto/signature');
+var Sighash = require('../sighash');
+var PublicKey = require('../../publickey');
+var BufferUtil = require('../../util/buffer');
+var TransactionSignature = require('../signature');
+
+/**
+ * @constructor
+ */
+function ScriptHashInput(input, pubkeys, redeemScript) {
+  Input.apply(this, arguments);
+  var self = this;
+  this.publicKeys = pubkeys || input.publicKeys;
+  this.threshold = 1;
+  this.redeemScript = redeemScript;
+  $.checkState(Script.buildScriptHashOut(this.redeemScript).equals(this.output.script),
+    'Provided redeemScript doesn\'t hash to the provided output');
```

Comment:
We [should use backslashes](https://github.com/airbnb/javascript#strings--escaping) to delimit the string and avoid escaping.